language/testing-infra/transactional-test-runner/src/framework.rs

```diff
@@ -389,41 +419,81 @@ where
     FIntoKnown: Fn(
         Command,
     ) -> KnownCommandFormat<
+        Adapter::ExtraInitArgs,
         Adapter::ExtraPublishArgs,
         Adapter::ExtraRunArgs,
         Adapter::Subcommand,
     >,
 {
-    // TaskOptions::from_args();
+    let extension = path.extension().unwrap().to_str().unwrap();
+    let default_syntax = if extension == MOVE_IR_EXTENSION {
+        SyntaxChoice::IR
+    } else {
+        assert!(extension == MOVE_EXTENSION);
+        SyntaxChoice::Source
+    };
     let mut output = String::new();
-    let tasks = taskify::<Command>(path).unwrap();
+    let mut tasks = taskify::<Command>(path)
+        .unwrap()
+        .into_iter()
+        .map(|t| t.map(&view_command))
+        .collect::<VecDeque<_>>();
+    assert!(!tasks.is_empty());
     let num_tasks = tasks.len();
     output.push_str(&format!(
         "processed {} task{}\n",
         num_tasks,
         if num_tasks > 1 { "s" } else { "" }
     ));
-    for task_input in tasks {
-        let task_number = task_input.number;
-        let task_name = task_input.name.to_owned();
-        let start_line = task_input.start_line;
-        let stop_line = task_input.stop_line;
-        let result = adapter.handle_command(task_input.map(&view_command));
-        let result_string = match result {
-            Ok(None) => continue,
-            Ok(Some(s)) => s,
-            Err(e) => format!("Error: {}", e),
-        };
-        assert!(!result_string.is_empty());
-        output.push_str(&format!(
-            "\ntask {} '{}'. lines {}-{}:\n{}\n",
-            task_number, task_name, start_line, stop_line, result_string
-        ));
+
+    let first_task = tasks.pop_front().unwrap();
+    let init_opt = match &first_task.command {
+        KnownCommandFormat::Init(_, _) => Some(first_task.map(|known| match known {
+            KnownCommandFormat::Init(command, extra_args) => (command, extra_args),
+            _ => unreachable!(),
+        })),
+        _ => {
+            tasks.push_front(first_task);
+            None
+        }
+    };
```

Comment:
Nit: I wonder if instead of using a VecDeque and actually moving the in and out of the container, can we just use slices?
```
let non_init_tasks = match tasks.get(0) {
    Some(task) => {
        process_init(task);
        &tasks[1..] 
    }
    None => &tasks[..]
}
```

Reply:
I need the owned values of the tasks for moving them into `KnownCommandFormat`